readme.md

```diff
@@ -76,109 +266,206 @@ specify the `--config` argument to Bazel with one of the following values:
+   ```
+5. Finally, call the `Run()` function, which will block indefinitely and receive data, calling the callback function you
+   provided each time data is received.
+   ```c++
+   client.Run();
+   ```
+6. When finished, call `Disconnect()` from somewhere in your code to disconnect from Polaris and return from `Run()`.
+   ```c++
+   client.Disconnect();
+   ```
+
+The `Run()` function handles authentication, connecting to Polaris, sending position updates, and receiving data. It is
+also responsible for handling error conditions including automatically reconnecting or reauthenticating as needed.
+
+If desired, you can use the `RunAsync()` function to launch `Run()` in a separate thread, returning control to your
+function immediately.
+
+### Example Applications ###
+
+#### Simple Polaris Client ####
+
+A small example of establishing a Polaris connection and receiving RTCM corrections data.
+
+#### Generic Serial Receiver Example ####
+
+This example relays incoming corrections data to a receiver over a serial connection. The receiver should be configured
+to send NMEA `$GPGGA` position updates over the connection. They will then be forwarded to Polaris to associate the
+receiver with an appropriate corrections stream.
+
+To run the application, run the following command:
+```
+bazel run //examples:serial_port_example -- --polaris_api_key=<POLARIS_API_KEY> --device=/dev/ttyACM0
+```
 
-### NTRIP Proxy Example ###
+#### NTRIP Proxy Example ####
 
-The example binary `ntrip_example` proxies Polaris RTK corrections as a simple NTRIP caster for use with devices that take NTRIP corrections.
+Receive incoming Polaris corrections data and relay it as a simple NTRIP caster for use with devices that take NTRIP
+corrections. Receiver positions sent to the NTRIP server (via a NMEA `$GPGGA` message) will be forwarded to Polaris to
+associate the receiver with an appropriate corrections stream.
 
-Any receiver that uses a NTRIP connection can simply connect to the computer running this application on port 2101 using the following command:
+For example, to run an NTRIP proxy on TCP port 2101 (the standard NTRIP port), run the following command:
 ```
-bazel run -c opt examples/ntrip:ntrip_example -- --polaris_api_key=${MY_POLARIS_APP_KEY} 0.0.0.0 2101 examples/ntrip
+bazel run -c opt examples/ntrip:ntrip_example -- --polaris_api_key=<POLARIS_API_KEY> 0.0.0.0 2101 examples/ntrip
 ```
 
-Your receiver should be configured to send NMEA GGA messages at a regaulr cadence (usually 1hz) so it is properly receiving corrections for the right region. The default mount point for the examples is `/Polaris` and no username or password is required.
+Any GNSS receiver that supports an NTRIP connection can then connect to the computer running this application to receive
+corrections, connecting to the NTRIP endpoint `/Polaris`. The receiver should be configured to send NMEA `$GPGGA`
+messages at a regular cadence (typically 1hz).
```

Comment:
We should also make a note that only one device (or really one location) can connect to the NTRIP server  at a time.